talk.tex

```diff
@@ -10,18 +10,152 @@
 
 \begin{document}
 
+
 \begin{frame}
 \maketitle
 \end{frame}
 
+
 \begin{frame}{Outline}
 \begin{block}{SymEngine}
 \begin{itemize}
-\item 1
-\item 2
-\item 3
+\item Introduction
+\item Why C++
+\item Internals of SymEngine
+\item SymPy, SymEngine and the interface
+\item Benchmarks
+\item Examples of usage
 \end{itemize}
 \end{block}
 \end{frame}
 
+
+\begin{frame}
+\frametitle{Introduction}
+\framesubtitle{About SymEngine}
+\begin{itemize}
+\item Symbolic manipulation library written in C++
+\item Started in 2012
+\item 35 contributors
+\item Part of SymPy organization
+\item MIT licensed
+\end{itemize}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Introduction}
+\framesubtitle{Goals}
+\begin{itemize}
+\item Be the fastest in symbolic manipulation (opensource or commercial)
+\item Serve as the core for SymPy
+\end{itemize}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Why C++}
+\begin{itemize}
+\item SymPy speed is insufficient to handle very large expressions.
+\item Easier to create bindings to other languages like Python, Julia, Ruby and Haskell.
+\end{itemize}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Internals of SymEngine}
+\begin{itemize}
+\item Depends on GNU Multiple Precision library (GMP)
+\item Optional dependencies include MPFR, MPC, Flint and Piranha
+\end{itemize}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Internals of SymEngine}
+\end{frame}
+
+
+
+\begin{frame}[fragile]
+\frametitle{SymPy, SymEngine and the interface}
+\begin{itemize}
+\item
+\begin{verbatim}
+>>> from sympy.core.backend import symbols, sin, diff
+\end{verbatim}
+\end{itemize}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Benchmarks}
+\framesubtitle{Modified GiNaC benchmark}
```

Comment:
@abhinavagarwal07, since you know about expand6b, can you write what the benchmark is in point form? See GiNaC paper which describes expand6